### PR TITLE
Use TIdentifier for RideId

### PR DESCRIPTION
**Pull request by ZehMatt**

### CODE REVIEW

File: src/openrct2/network/NetworkTypes.h [hunk 1]
```diff
@@ -126,7 +126,7 @@ template<typename T, size_t _TypeID> struct NetworkObjectId_t
 };
 #pragma pack(pop)
 
-using NetworkRideId_t = ride_id_t;
+using NetworkRideId_t = RideId;
```
Comment (duncanspumpkin) [inline src/openrct2/network/NetworkTypes.h:129]: Why is we need a separate NetworkRideId_t type?

File: src/openrct2/park/ParkFile.cpp [hunk 2]
```diff
@@ -1516,9 +1516,9 @@ namespace OpenRCT2
                         return true;
                     });
                     cs.Ignore<uint64_t>();
-                    cs.Ignore<ride_id_t>();
-                    cs.Ignore<ride_id_t>();
-                    cs.Ignore<ride_id_t>();
+                    cs.Ignore<RideId>();
```
Comment (duncanspumpkin) [inline src/openrct2/park/ParkFile.cpp:1519]: I wonder if these should perhaps become `uint16_t`'s just thinking about in a future where RideId is not `uint16_t`

File: src/openrct2/network/NetworkTypes.h (hunk 1 above)
Comment (ZehMatt) [reply src/openrct2/network/NetworkTypes.h:129]: We don't, it used to be a different thing prior to ride_id_t becoming enum.

File: src/openrct2/park/ParkFile.cpp (hunk 2 above)
Comment (ZehMatt) [reply src/openrct2/park/ParkFile.cpp:1519]: I wonder, currently we have rides capped at 1000, so that makes only 1.5% of used space. I wouldn't mind a static assert so we don't forget when we happen to change it.

File: src/openrct2/ride/RideRatings.cpp [hunk 3]
```diff
@@ -160,20 +160,20 @@ static void ride_ratings_update_state(RideRatingUpdateState& state)
  */
 static void ride_ratings_update_state_0(RideRatingUpdateState& state)
 {
-    ride_id_t currentRide = state.CurrentRide;
+    auto currentRide = state.CurrentRide;
 
-    currentRide = static_cast<ride_id_t>(EnumValue(currentRide) + 1);
-    if (currentRide >= static_cast<ride_id_t>(MAX_RIDES))
+    auto nextRide = RideId::FromUnderlying(currentRide.ToUnderlying() + 1);
```
Comment (Gymnasiast) [inline src/openrct2/ride/RideRatings.cpp:165]: Perhaps it would be a good idea to implement `operator++` for TIdentifier?

File: src/openrct2-ui/windows/NewCampaign.cpp [hunk 4]
```diff
@@ -27,7 +27,7 @@ static constexpr const rct_string_id WINDOW_TITLE = STR_NONE;
 static constexpr const int32_t WH = 109;
 static constexpr const int32_t WW = 350;
 
-constexpr auto SELECTED_RIDE_UNDEFINED = RIDE_ID_NULL;
+constexpr auto SELECTED_RIDE_UNDEFINED = RideId::GetNull();
```
Comment (Broxzier) [inline src/openrct2-ui/windows/NewCampaign.cpp:30]: This could be dropped. `Campaign::rideId` is of the same type, so this doesn't add anything extra.

File: src/openrct2/actions/ParkMarketingAction.cpp [hunk 5]
```diff
@@ -70,7 +70,7 @@ GameActions::Result ParkMarketingAction::Execute() const
     campaign.Flags = MarketingCampaignFlags::FIRST_WEEK;
     if (campaign.Type == ADVERTISING_CAMPAIGN_RIDE_FREE || campaign.Type == ADVERTISING_CAMPAIGN_RIDE)
     {
-        campaign.RideId = static_cast<ride_id_t>(_item);
+        campaign.rideId = RideId::FromUnderlying(_item);
```
Comment (Broxzier) [inline src/openrct2/actions/ParkMarketingAction.cpp:73]: What about changing `_item` to be `RideId`?

Comment (ZehMatt) [reply src/openrct2/actions/ParkMarketingAction.cpp:73]: Because its not always a ride id.

Comment (Broxzier) [reply src/openrct2/actions/ParkMarketingAction.cpp:73]: Alright, out of scope for this PR then.

File: src/openrct2/ride/RideRatings.cpp (hunk 3 above)
Comment (ZehMatt) [reply src/openrct2/ride/RideRatings.cpp:165]: I don't think this instance warrants that, I rather have this code cleaned up to use the RideManager and its iterators, they already implement this. Definitely for a future PR

File: src/openrct2-ui/windows/NewCampaign.cpp [hunk 6]
```diff
@@ -173,7 +172,7 @@ class NewCampaignWindow final : public Window
         campaign.no_weeks = 2;
 
         // Currently selected ride
-        campaign.RideId = SELECTED_RIDE_UNDEFINED;
+        campaign.rideId = RideId::GetNull();
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/NewCampaign.cpp:175]: Why did you change this TitleCase to camelCase? We are moving towards the former.

Comment (Broxzier) [reply src/openrct2-ui/windows/NewCampaign.cpp:175]: Hence the note about using `_t` suffixes for types or `m_` prefixes for data members. There's a whole lot, but changing them all is something clang-tidy could probably do.

Comment (ZehMatt) [reply src/openrct2-ui/windows/NewCampaign.cpp:175]: Adding _t to every type makes the code look a bit strange over time, this means every custom type should have that. We probably have to go over our own code style again as it has problematic aspects using PascalCase for public members. So where does this leave us with this PR?

File: src/openrct2/park/ParkFile.cpp (hunk 2 above)
Comment (ZehMatt) [reply src/openrct2/park/ParkFile.cpp:1519]: I think its better to handle this when the time comes to, then we simply use the park version.

File: src/openrct2-ui/windows/NewCampaign.cpp (hunk 6 above)
Comment (Broxzier) [reply src/openrct2-ui/windows/NewCampaign.cpp:175]: For this PR, it's probably best to stick with our current style with PascalCase, and then use `::RideId` where you need to access the variable instead of the type.

Comment (ZehMatt) [reply src/openrct2-ui/windows/NewCampaign.cpp:175]: I don't think this will work but I'll give it a try on the CI.

Comment (ZehMatt) [reply src/openrct2-ui/windows/NewCampaign.cpp]: All good now.

**Review by Gymnasiast — APPROVED** [review]
